omc3/check_corrections.py

```diff
@@ -0,0 +1,612 @@
+        _do_plots(corrections, opt)
+
+
+# Input Parameters -------------------------------------------------------------
+
+def _check_opt_add_dicts(opt: DotDict) -> DotDict:  # acts inplace...
+    """ Check on options and put in missing values. """
+    def_dict = _get_default_values()
+
+    # Check cuts and fill defaults
+    if opt.optics_params:
+        for key in ("modelcut", "errorcut"):
+            if opt[key] is not None and len(opt[key]) != len(opt.optics_params):
+                raise AttributeError(f"Length of {key} is not the same as of the optical parameters!")
+            if opt[key] is not None:
+                opt[key] = dict(zip(opt.optics_params, opt[key]))
+            else:
+                opt[key] = {param: def_dict[key][param] for param in opt.optics_params}
+
+    # Convert Strings to Paths
+    opt.meas_dir = Path(opt.meas_dir)
+    opt.output_dir = Path(opt.output_dir)
+    opt.corrections = [Path(c) for c in opt.corrections]
+    all_files = all(c.is_file() for c in opt.corrections)
+    all_dirs = all(c.is_dir() for c in opt.corrections)
+    if not (all_files or all_dirs):
+        raise AttributeError("Corrections need to be either all paths to files or all paths to directories!")
+
+    if all_dirs and not opt.file_pattern:
+        raise AttributeError("Parameter 'file_pattern' is missing, when giving directories as input.")
+
+    return opt
+
+
+def _get_corrections(corrections: Sequence[Path], file_pattern: str = None) -> Dict[str, Sequence[Path]]:
+    """ Sort the given correction files:
+    If given by individual files, they all go into one bucket,
+    if given by folders (i.e. scenarios) they are sorted by its name.
+    It is also checked, that they are valid!"""
+    # create correction mapping
+    if corrections[0].is_file():  # checked above, that all are files or all are dirs
+        corr_dict = {"": corrections}
+    else:
+        corr_dict = {c.name:  _glob_regex_paths(c, file_pattern) for c in corrections}
+
+    # check correction files
+    for name, corr_files in corr_dict.items():
+        if not len(corr_files):
+            raise IOError(f"No corrections found for scenario {name}.")
+
+        do_not_exist = [f for f in corr_files if not f.exists()]
+        if len(do_not_exist):
+            raise IOError(f"Some correction files do not exist for scenario {name}:"
+                          f" {str(do_not_exist)}")
+
+    return corr_dict
+
+
+def _glob_regex_paths(path: Path, pattern: str) -> List[Path]:
+    """ Filter the files in path by pattern and return a list of paths. """
+    return [path / f for f in glob_regex(path, pattern)]
+
+
+# Main and Output --------------------------------------------------------------
+
+def _get_measurement_filter(nominal_model: TfsDataFrame, opt: DotDict) -> Dict[str, pd.Index]:
+    """ Writes the filtered original data out, so that it can be easily plotted against
+    the corrections later. We now have the same BPMs and Column-Names. """
```

Comment:
If I understand correctly: this loads the measurements for different optics properties from the measurement folder, then applies a filter based on the params in `opt` and makes sure we have the same index (so all in all, the same common filtering) for all of them?

I would like to have this explanation in the docstring, even though it's a private function.

Reply:
I think I accidentally copied this docstring from somewhere. This docstring makes no sense for this function

Reply:
(or maybe it was an old version of this function)